### PR TITLE
chore: move networkmonitor and wakucanary to apps directory

### PR DESCRIPTION
**Pull request by LNSD**
This PR brings a simplification to the repository structure and the removal of a long time outdated and deprecated files and scripts:

- [x] Moved `networkmonitor` and `wakucanary` sources under the `apps` folder.
- [x] Removed the `tools/scripts` and `tools/simulations` outdated and deprecated files.
- [x] Removed the `tools` directory in favor of `apps`.
- [x] Updated the nimble script and the Makefile accordingly. 

These changes aim to simplify the imports' cognitive load. These are some of the rules that will prevent future code issues (e.g., cyclic import issues):

- Apps should be self-contained: An app MUST NOT import code from another app.
- Apps CAN import code from the libraries folder (e.g., `waku/common`, `waku/v1`, or `waku/v2`).
- Libraries (code under `waku`) MUST NOT import code from the apps.

### CODE REVIEW

**Review by rymnc — APPROVED** [review]
LGTM

**Review by vpavlin — APPROVED** [review]
LGTM

File: waku.nimble [hunk 1]
```diff
@@ -116,11 +92,24 @@ task chat2bridge, "Build chat2bridge":
...
 
-### Waku Tooling
-task wakucanary, "Build waku-canary tool":
-  let name = "wakucanary"
-  buildBinary name, "tools/wakucanary/", "-d:chronicles_log_level=TRACE"
 
-task networkmonitor, "Build network monitor tool":
-  let name = "networkmonitor"
-  buildBinary name, "tools/networkmonitor/", "-d:chronicles_log_level=TRACE"
+### Legacy: Whisper & Waku v1 tasks
```
Comment (Ivansete-status) [inline waku.nimble:96]: Shall we remove this ?

**Review by Ivansete-status — APPROVED** [review]

File: waku.nimble (hunk 1 above)
Comment (LNSD) [reply waku.nimble:96]: We have not yet deprecated Waku v1. So I cannot remove legacy code and the associated tasks.

**Pull request merged**